### PR TITLE
Adding Perp methods

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: provider/http.go [hunk 1]
```diff
@@ -126,6 +127,23 @@ func (h *HTTPClient) GetOpenOrders(ctx context.Context, market string, owner str
...
 }
 
+// GetPerpPositions returns all perp positions by owner address and market
+func (h *HTTPClient) GetPerpPositions(ctx context.Context, ownerAddress string, accountAddress string, contracts []common.PerpContract, project pb.Project) (*pb.GetPerpPositionsResponse, error) {
+	var strs []string
+	for _, c := range contracts {
+		strs = append(strs, fmt.Sprint(c))
+	}
+
+	contractsArg := convertStrSliceArgument("contracs", true, strs)
```
Comment (aspin) [inline provider/http.go]: contracs => contracts

Comment (tinystarinagalaxy) [reply provider/http.go]: done

File: provider/http.go [hunk 2]
```diff
@@ -377,6 +435,38 @@ func (h *HTTPClient) SubmitCancelOrder(
...
 }
 
+// PostClosePerpPositions builds cancel perp positions txn.
+func (h *HTTPClient) PostClosePerpPositions(ctx context.Context, ownerAddress string, contracts []common.PerpContract, project pb.Project) (*pb.PostClosePerpPositionsResponse, error) {
+	request := &pb.PostClosePerpPositionsRequest{
+		Project:      project,
+		OwnerAddress: ownerAddress,
+		Contracts:    contracts,
+	}
+	url := fmt.Sprintf("%s/api/v1/trade/close-perp-positions", h.baseURL)
```
Comment (aspin) [inline provider/http.go:445]: make sure you update these strings

Comment (aspin) [reply provider/http.go:445]: this one's `trade/perp/close-positions` i think

File: provider/http.go [hunk 3]
```diff
@@ -126,6 +127,23 @@ func (h *HTTPClient) GetOpenOrders(ctx context.Context, market string, owner str
...
 
+// GetPerpPositions returns all perp positions by owner address and market
+func (h *HTTPClient) GetPerpPositions(ctx context.Context, ownerAddress string, accountAddress string, contracts []common.PerpContract, project pb.Project) (*pb.PerpPositionsResponse, error) {
+	var strs []string
+	for _, c := range contracts {
+		strs = append(strs, fmt.Sprint(c))
+	}
+
+	contractsArg := convertStrSliceArgument("contracts", true, strs)
+	url := fmt.Sprintf("%s/api/v1/trade/perp-positions/?ownerAddress=%s&accountAddress=%s&project=%s%s", h.baseURL, ownerAddress, accountAddress, project, contractsArg)
```
Comment (aspin) [inline provider/http.go:138]: same here

**Review by aspin — APPROVED** [review]
couple of http endpoint need to updated per recent proto change feedback, otherwise LGTM

**Pull request merged**